providers/core.tcpsocket.js

```diff
@@ -517,6 +522,11 @@ Socket_chrome.handleReadData = function (readInfo) {
  * @static
  */
 Socket_chrome.handleReadError = function (readInfo) {
+  if (readInfo.resultCode === -100) {
```

Comment:
This doesn't fix the uProxy integration tests.

Digging down a little, it seems like `handleReadError` is being called with -100 whenever a socket terminates - maybe even before the `close` callback.

I created a commit to help debug this:
https://github.com/uProxy/uproxy/compare/trevj-tcp-socket-close-debugging?expand=1

Can you patch this in and experiment a little?